### PR TITLE
Terminate when map creation fails

### LINKED ISSUE #729: Don't try to attach to probes if failed to attach to BPF maps
From https://github.com/iovisor/bpftrace/issues/664, bpftrace seems to try to attach to probes even after it failed to create BPF maps. If we couldn't create maps, bpftrace should exit gracefully and with a suggestive error message (for example, suggesting the user to run `dmesg`, there will probably be clues of what happened there).

### PR DESCRIPTION
**Pull request by fbs**
When creating a map fails bpftrace needs to be terminated to prevent undefined/wrong behaviour (usually segv).

```
./src/bpftrace -e 'i:s:1 { @=1; @b=2; }'
Error creating map: '@': Resource temporarily unavailable
Error creating map: '@b': Resource temporarily unavailable
Error creating printf map (-1): Resource temporarily unavailable
Creation of the required BPF maps has failed.
Make sure you have all the required permissions and are not confined (e.g. like
snapcraft does).  `dmesg` will likely have useful output for further troubleshooting
````

This fixes #729

It was tested by inserting the following bit of code into main:

```
scmp_filter_ctx ctx;
ctx = seccomp_init(SCMP_ACT_ALLOW);
seccomp_rule_add(ctx, SCMP_ACT_ERRNO(11), SCMP_SYS(bpf), 0);
seccomp_load(ctx);
```

### CODE REVIEW

**Comment by fbs** [comment]
To make testing a bit easier I wrote a simple seccomp wrapper that can block specific bpf calls:

```
sudo ./seccomp -e map_create:1   -- ./src/bpftrace -e 'i:s:1 { @=5 }'
Added ERRNO(1) for command: map_create(0)
Executing: ./src/bpftrace -e i:s:1 { @=5 }
------------

Error creating map: '@': Operation not permitted
```

I've added it to the scripts directory as it's useful for testing failures. Any objections to this?

**Comment by danobi** [comment]
Seems reasonable, but curious about use case. Is this when bpftrace is run inside a container or something?

**Comment by fbs** [comment]
The issue I've seen most is snapcraft blocking some or all calls which cause weird errors like the one here. Limited syscall access in containers could be another case but I'm not familiar with that.

File: src/map.cpp [hunk 1]
```diff
@@ -117,21 +119,12 @@ Map::Map(enum bpf_map_type map_type)
...
-      case BPF_MAP_TYPE_PERF_EVENT_ARRAY:
-        name = "perf event";
-        break;
-      default:
-        std::cerr << "invalid map type" << std::endl;
-        abort();
-    }
-
-    std::cerr << "Error creating " << name << " map (" << mapfd_ << ")" << std::endl;
+    std::cerr << "Error creating " << name << " map: " << strerror(errno);
```
Comment (danobi) [inline src/map.cpp]: Can combine this into one `std::cerr` call like above.

File: scripts/seccomp.c [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include <linux/bpf.h>
+#include <seccomp.h>
+
+// Compile with gcc /vagrant/scripts/seccomp.c -lseccomp -ggdb -o seccomp
+
+void help() {
```
[the remark this answers is not in the document]
Comment (fbs) [reply scripts/seccomp.c]: Could try. Not sure it improves readability though.

**Comment by fbs** [comment]
> Next time could you put the formatting changes in a separate commit or PR? Makes it easier to review.

Will do. I removed the useless extra formatting from this MR to keep the diffs readable.

**Pull request merged**